### ISSUE TITLE
Implement Regression Capability

### ISSUE THREAD

**Issue by DeepBlockDeepak** [DeepBlockDeepak/my-neural-network#17]
## Problem
Current architecture only allows for classification (bin, multi). Regression capabilities are needed for more tasks.

## Ideas/Steps
- Include **task_type** or something else, for a Config parameter. Distinguish for "classification" or "regression"
- Modify output layer activation function in **forward_propagation()** to respond to the task_type. Needs to include linear activation for using regression (no activation function required).
- ? incorporate MSE for the compute loss function & finally incorporate **compute_loss()**  in logic as well, since it's not used?
- Make some kind of change to backprop? Loss should be the same as in classifcation tasks, so maybe not needed?
- Modify predict() so that it knows to return the AL? Does a max() need to be applied?
